components/raftstore/src/store/fsm/peer.rs

```diff
@@ -3283,17 +3299,32 @@ where
             return;
         }
 
-        if let Err(e) = self.check_merge_proposal(&mut msg) {
-            warn!(
-                "failed to propose merge";
-                "region_id" => self.region_id(),
-                "peer_id" => self.fsm.peer_id(),
-                "message" => ?msg,
-                "err" => %e,
-                "error_code" => %e.error_code(),
-            );
-            cb.invoke_with_response(new_error(e));
-            return;
+        if msg.has_admin_request() {
+            if let Err(e) = self.check_joint_state(&msg) {
+                warn!(
```

Comment:
Does PD follows the same constraint?

Reply:
Yes, when a region in joint state PD won't try to split/merge this region /cc @HunDunDM